app/dashboard/src/hooks/syncRefHooks.ts

```diff
@@ -12,11 +12,12 @@ import * as React from 'react'
 export function useSyncRef<T>(value: T): Readonly<React.MutableRefObject<T>> {
   const ref = React.useRef(value)
 
-  // Update the ref value whenever the provided value changes
-  // Refs shall never change during the render phase, so we use `useEffect` here.
-  React.useEffect(() => {
-    ref.current = value
-  })
+  /*
+  Even though the react core team doesn't recommend setting ref values during the render (it might lead to deoptimizations), the reasoning behind this is:
```

Comment:
nits:
- `react` -> `React`
- also i prefer wrapping code in backticks (`useEventCallback`, `const x = () => {}`, `useCallback`)
  - not present here but for things that are *not* code i would prevent \*italics\* or \*\*bold\*\* for emphasis
- i think this line is too long? (> 100 characters)
- block comments should ideally have leading `*` on each line. i guess nowadays it doesn't matter because of highlighting, but i don't think it *hurts* to add them anyway